compiler/include/type.h

```diff
@@ -152,9 +152,16 @@ enum AggregateTag {
   AGGREGATE_UNION
 };
 
+enum InstantiationStyle {
+  DEFINES_CONSTRUCTOR,
+  DEFINES_INITIALIZER,
+  DEFINES_NONE_USE_DEFAULT
+};
+
 class AggregateType : public Type {
  public:
   AggregateTag aggregateTag;
+  InstantiationStyle instantiationStyle;
```

Comment:
I think it'd be better to call this "initializerStyle". In a forward-looking way, that's still reasonable, and it avoids the new meaning of instantiation (where normally I would think about what happens to generic functions in resolution as "instantiation").